UNIXFS.md

```diff
@@ -59,6 +63,76 @@ For files that are comprised of more than a single block, the 'Type' field will
 
 This data is serialized and placed inside the 'Data' field of the outer merkledag protobuf, which also contains the actual links to the child nodes of this object.
 
+## Metadata
+
+UnixFS currently supports three metadata fields:
+
+* `mimeType` -- The mime-type of the file. This generally shouldn't be used.
+* `mode` -- The `mode` is for optionally persisting the [file permissions in numeric notation](https://en.wikipedia.org/wiki/File_system_permissions#Numeric_notation) \[[spec](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/sys_stat.h.html)\]. It defaults to 0755 if unspecified.
+* `mtime` -- The modification time in seconds since the epoch. This defaults to the unix epoch if unspecified.
+
+There are two ways to specify file metadata:
+
+### Embedded in the directory
+
+Each entry in the repeated metadata field corresponds to the linked file/directory at the same offset in the Links section of the outer dag-pb. This field is appropriate in nodes with type `Directory` and `HAMTShard`. However, metadata corresponding to linked HAMT shards should be left blank.
+
+For example, a directory might look like.
+
+```
+{
+  Type: Directory,
+  Data: {
+    Metadata: [
+      {},
+      {Mode: 0666},
+      {}
+  },
+  Links: [
+    {Name: "foo", ...}, // mode: 0644 (default)
```

Comment:
Above we say that the mode `defaults to 0755 if unspecified` but here we say `mode: 0644 (default)`